hbase-server/src/test/java/org/apache/hadoop/hbase/io/hfile/TestPrefetch.java

```diff
@@ -295,10 +302,23 @@ private void readStoreFile(Path storeFilePath,
     throws Exception {
     // Open the file
     HFile.Reader reader = HFile.createReader(fs, storeFilePath, cacheConfig, true, conf);
+    startTimer();
 
     while (!reader.prefetchComplete()) {
       // Sleep for a bit
       Thread.sleep(1000);
+      if (getComputeTiming()) {
+        // After task is scheduled and before the delay expires, prefetch should not start
+        // if prefetchFutures contains entry (which means it's not cancelled or completed)
+        // and wait time remaining is below delay expiry watermark, it can be deduced that
+        // the prefetch is not started yet.
+        if (getElapsedTime() >= (conf.getLong(PREFETCH_DELAY, 1000))) {
+          assertTrue("Prefetch should be started at this point", reader.prefetchStarted());
+          setComputeTiming(false);
+        } else {
+          assertFalse("Prefetch Should not start at this point", reader.prefetchStarted());
+        }
+      }
```

Comment:
This version of readStoreFile enforces the prefetch execution and completion. This is relevant to some of the tests in this class, but definitely not for yours. 

You should rather just move the reader creation line to your test method, then put your validation logic there. That would yield the following benefits:
1) Avoid the need for the additional global variables that are only of concern of a single test method (anti pattern);
2) Make your test easier to be understood by other readers;
3) Make your test more cohesive, and quicker, as it doesn't need to wait for the whole prefetch completion. 
4) Keep existing version of readStoreFile more cohesive, as it doesn't implement additional variations of behaviour.

Reply:
In prefetch executor, we further compute passed in delay using variation and a random
    multiplier to get 'effective delay'. Hence, in the test, for delay of 25000 milli-secs
    checking that prefetch is started after 20000 milli-sec and prefetch started after that.
    However, prefetch should not start after configured delay.

    In previous change set, I had introduced new class variable i.e. computedDealy which could
    have helped to get the exact value however, later removed due to previous comments.

    Tried to disable the PREFETCH_DELAY_VARIATION in the tests but looks like it's not possible.